[Identity] DefaultAzureCredential clarification (#13830)
Azure/azure-sdk-for-js: sdk/identity/identity/README.md

### DefaultAzureCredential

> The `DefaultAzureCredential` is appropriate for most scenarios where the application is intended to ultimately be run in the Azure Cloud. This is because the `DefaultAzureCredential` combines credentials commonly used to authenticate when deployed, with credentials used to authenticate in a development environment.

Reviewer: ```suggestion
The `DefaultAzureCredential` is appropriate for most scenarios where the application is intended to ultimately be run in the Azure Cloud. This is because the `DefaultAzureCredential` combines credentials commonly used to authenticate when deployed with credentials used to authenticate in a development environment.
```

> Keep in mind that while `DefaultAzureCredential` is intended to simplify getting started with the SDK by handling common scenarios with reasonable default behavior, developers who want more control, or whose scenario isn't served by the defaults we chose, should use other credential types.

Reviewer: ```suggestion
> Note: `DefaultAzureCredential` is intended to simplify getting started with the SDK by handling common scenarios with reasonable default behaviors. Developers who want more control or whose scenario isn't served by the default settings should use other credential types.
```

Author: Much better! Thanks!
